hyperspy/gui/egerton_quantification.py

```diff
@@ -369,10 +369,12 @@ def _interpolator_kind_changed(self, old, new):
         self.span_selector_changed()
 
     def _ss_left_value_changed(self, old, new):
-        self.span_selector_changed()
+        if not np.isnan(self.ss_right_value):
+            self.span_selector_changed()
 
     def _ss_right_value_changed(self, old, new):
-        self.span_selector_changed()
+        if not np.isnan(self.ss_right_value):
```

Comment:
Should this check `self.ss_left_value`? If not, maybe it should check `new` instead?